src/vscode-dts/vscode.d.ts

```diff
@@ -19654,6 +19737,306 @@ declare module 'vscode' {
 		 */
 		canSendRequest(chat: LanguageModelChat): boolean | undefined;
 	}
+
+	/**
+	 * A tool that is available to the language model via {@link LanguageModelChatRequestOptions}. A language model uses all the
+	 * properties of this interface to decide which tool to call, and how to call it.
+	 */
+	export interface LanguageModelChatTool {
+		/**
+		 * The name of the tool.
+		 */
+		name: string;
+
+		/**
+		 * The description of the tool.
+		 */
+		description: string;
+
+		/**
+		 * A JSON schema for the parameters this tool accepts.
+		 */
+		parametersSchema?: object;
+	}
+
+	/**
+	 * A tool-calling mode for the language model to use.
+	 */
+	export enum LanguageModelChatToolMode {
+		/**
+		 * The language model can choose to call a tool or generate a message. Is the default.
+		 */
+		Auto = 1,
+
+		/**
+		 * The language model must call one of the provided tools. Note- some models only support a single tool when using this
+		 * mode.
+		 */
+		Required = 2
+	}
+
+	/**
+	 * A language model response part indicating a tool call, returned from a {@link LanguageModelChatResponse}, and also can be
+	 * included as a content part on a {@link LanguageModelChatMessage}, to represent a previous tool call in a chat request.
+	 */
+	export class LanguageModelToolCallPart {
+		/**
+		 * The name of the tool to call.
+		 */
+		name: string;
+
+		/**
+		 * The ID of the tool call. This is a unique identifier for the tool call within the chat request.
+		 */
+		callId: string;
+
+		/**
+		 * The parameters with which to call the tool.
+		 */
+		parameters: object;
+
+		/**
+		 * Create a new LanguageModelToolCallPart.
+		 */
+		constructor(name: string, callId: string, parameters: object);
+	}
+
+	/**
+	 * A language model response part containing a piece of text, returned from a {@link LanguageModelChatResponse}.
+	 */
+	export class LanguageModelTextPart {
+		/**
+		 * The text content of the part.
+		 */
+		value: string;
+
+		/**
+		 * Construct a text part with the given content.
+		 * @param value The text content of the part.
+		 */
+		constructor(value: string);
+	}
+
+	/**
+	 * A language model response part containing a PromptElementJSON from `@vscode/prompt-tsx`.
+	 * @see {@link LanguageModelToolResult}
+	 */
+	export class LanguageModelPromptTsxPart {
+		/**
+		 * The value of the part.
+		 */
+		value: unknown;
+
+		/**
+		 * The mimeType of this part, exported from the `@vscode/prompt-tsx` library.
+		 */
+		mime: string;
```

Comment:
The prompt-tsx contentType has a number that could be incremented as a version, so that's why I included it. But, we could also omit it for now and just add it in the future if we actually do that.

Reply:
(must have lost a comment, reposting and hopefully not double posting)

Should we have the type from `LanguageModelPromptTsxPart` to just `LanguageModelJSONPart` which value and an optional type?

Reply:
> The prompt-tsx contentType has a number that could be incremented as a version, 

We just need to be careful with supporting the forever, so tool A from extension A uses version 1 and the LM implemtator uses version 2